File: packages/sdk-js/src/resources/feedback.ts
import { EidolonDBClient } from '../client.js';
import type { MarkUsedResponse, RetrievalStatsResponse } from '../types.js';

export class FeedbackResource {
  constructor(private readonly client: EidolonDBClient) {}

  markUsed(
    memoryIds: string[],
    options?: {
      sessionId?: string;
      actorId?: string;
      relevanceFeedback?: Record<string, number>;
    }
  ): Promise<MarkUsedResponse> {
    return this.client.request<MarkUsedResponse>('POST', '/feedback/mark-used', {
      memoryIds,
      ...options,
    });
  }

  stats(memoryId: string): Promise<RetrievalStatsResponse> {
    return this.client.request<RetrievalStatsResponse>('GET', `/feedback/stats/${memoryId}`);
  }

  async listStats(options?: { limit?: number; sortBy?: string }): Promise<RetrievalStatsResponse[]> {
    const response = await this.client.request<{ stats: RetrievalStatsResponse[] }>(
      'GET',
      '/feedback/stats',
      undefined,
      { query: options }
    );
    return response.stats;
  }
}

File: packages/sdk-python/src/eidolondb/resources/feedback.py
from __future__ import annotations

from typing import Any, Dict, Optional

from .._client import EidolonDBClient
from .._types import MarkUsedResponse, RetrievalStatsResponse


class FeedbackResource:
    def __init__(self, client: EidolonDBClient):
        self._client = client

    def mark_used(
        self,
        memory_id: str,
        *,
        query: Optional[str] = None,
        retrieval_score: Optional[float] = None,
        relevance: Optional[float] = None,
        metadata: Optional[Dict[str, Any]] = None,
    ) -> MarkUsedResponse:
        payload = {
            "memoryId": memory_id,
            "query": query,
            "retrievalScore": retrieval_score,
            "relevance": relevance,
            "metadata": metadata,
        }
        return self._client.request("POST", "/feedback/mark-used", body=payload)

    def retrieval_stats(self) -> RetrievalStatsResponse:
        return self._client.request("GET", "/feedback/retrieval/stats")
